docs: Fix style pitfalls in the ClusterMesh guide (#32320)
cilium-team/cilium: Documentation/network/clustermesh/aks-clustermesh-prep.rst

AKS (Azure Kubernetes Service) clusters in BYOCNI mode to meet the requirements 
for the clustermesh feature.

> In this guide you will install two AKS clusters in BYOCNI (Bring Your Own CNI) 

Reviewer: ```suggestion
This guide describes how to install two AKS clusters in BYOCNI (Bring Your Own CNI) 
```

        #  westus2 can be changed to any available location (`az account list-locations`)
        az group create --name "${AZURE_RESOURCE_GROUP}" -l westus2

2.  Now that you have a resource group you can create a VNet (virtual network). 
    Creating a custom VNet is required so you can specify a unique Node, Pod, and 

>     Service CIDRs to make sure you don't overlap with other clusters.

Reviewer: ```suggestion
2.  Create a VNet (virtual network). 
    Creating a custom VNet is required to ensure that the Node, Pod, and 
    Service CIDRs are unique and they don't overlap with other clusters.
```

    .. note::

>         This case uses the ``192.168.10.0/24`` range, but this can be exchanged

Reviewer 2: Please make this note consistent with the other note on L128.

Author: Done

>         In this case you use the ``192.168.10.0/24`` range, but this can be exchanged

Reviewer: ```suggestion
        This case uses the ``192.168.10.0/24`` range, but this can be exchanged
```

2.  Create a VNet in this resource group. Make sure to use a non-overlapping prefix.

    .. note::

>         Use the ``192.168.20.0/24`` range, but this can be exchanged

Reviewer 2: This one sounds a bit confusing. We can't really give directives inside of a note, and “use this but you could also have that” sounds unclear. Maybe something like: “The example below uses range ``192...``, but you could use instead any range except for...”

Author: Done

Peering virtual networks
########################

Virtual networks can't connect to each other by default. You can enable cross
VNet communication by creating bi-directional "peering".

> Create a peering from cluster one to cluster two using the

Reviewer 2: (Nit: or “First, create ...”)

There's a remaining occurrence of “we” (“we need to add ...”) further down in this file, on line 221. Can you also fix it, please?

Author: Done

Author: I just realized it didn't save. I have fixed it.
